src/corppa/poetry_detection/annotation/process_adjudication_data.py

```diff
@@ -32,6 +32,24 @@
 from xopen import xopen
 
 
+def clean_excerpt(span: dict[str, int | str]):
+    """
+    Clean excerpt so that all leading and trailing whitespace is removed.
```

Comment:
Might be nice to document what the span dict fields are (maybe it's obvious in context of this file but I'd forgotten and we have other spans)
```suggestion
    Clean excerpt so that any leading and trailing whitespace is removed, and update start and end values to match the cleaned text. Expects a dictionary with start and end indices and text content.
```

Reply:
Sure, I can add slightly more info; since an excerpt is more than just a text string, it seemed clearly implied to me. But I'll expand on the comment (but make `ruff` happy)

Reply:
To make this more explicit, I made the Excerpt a `TypedDict` which makes it more explicit what these structures are and also makes typing hints more accurate.